tests/ignite/engine/test_engine.py

```diff
@@ -366,6 +367,40 @@ def test_state_get_event_attrib_value():
     assert state.get_event_attrib_value(e) == state.epoch
 
 
+def test_time_stored_in_state():
+    def _test(data, max_epochs, epoch_length):
+
+        engine = Engine(lambda e, b: time.sleep(0.01))
+
+        num_examples = len(data)
+
+        engine.run(data, max_epochs=max_epochs, epoch_length=epoch_length)
+
+        lower_bound_time = num_examples / epoch_length / max_epochs
+        upper_bound_time = 1.1 * lower_bound_time
+
+        def init(engine):
+            engine.state.times = {Events.EPOCH_COMPLETED.name: 0.0, Events.COMPLETED.name: 0.0}
```

Comment:
I think we should attach this automatically without explicitly require the user

Reply:
Excellent. This initialization happens in the `State` constructor, but I forgot to remove this from the test; that said, I think that's slightly different that wiring up the event handler.

Reply:
Is there some place where it makes sense to wire this or is `engine.run` good enough?